src/wallet/wallettool.cpp

```diff
@@ -141,10 +141,12 @@ bool ExecuteWalletToolFunc(const std::string& command, const std::string& name)
             return false;
         }
         bilingual_str error;
-        if (!WalletBatch::VerifyEnvironment(path, error)) {
+        std::unique_ptr<WalletDatabase> database = CreateWalletDatabase(path);
+        if (!database->VerifyNotInUse(error)) {
```

Comment:
In commit "walletdb: Combine VerifyDatabaseFile and VerifyEnvironment" (cba81c3272405332cbb2f3b9c6b2c828c1869d16)

> I've added `VerifyNotInUse`. This function is still called by `Verify` so that the behavior of other `Verify` calls here won't change.

I think this open before open code in wallet tool can be dropped entirely. It seems like the only purpose of this is to show an "Error loading %s. Is wallet being used by other process?" string instead of an "Error loading wallet. %s" string

Reply:
I've dropped this check and added the `env->Open` that `SalvageWallet` needs to `SalvageWallet`.